org.opent2t.onboarding.hue/js/thingOnboarding.js

```diff
@@ -0,0 +1,175 @@
+                    disgestHeaderContent += 'nonce=\"' + nonce + '\", ';
+                    disgestHeaderContent += 'uri=\"/oauth2/token\", ';
+                    var HASH1 = crypto.createHash('md5').update(authInfo[0].client_id + ':oauth2_client@api.meethue.com:' + authInfo[0].client_secret).digest('hex');
+                    var HASH2 = crypto.createHash('md5').update('POST:/oauth2/token').digest('hex');
+                    var authHeaderResponse = crypto.createHash('md5').update(HASH1 + ':' + nonce + ':' + HASH2).digest('hex');
+                    disgestHeaderContent += 'response=\"' + authHeaderResponse + '\"';
+
+                    options.headers = {
+                        'Accept': 'application/json',
+                        'Authorization': 'Digest ' + disgestHeaderContent,
+                    };
+
+                    return request(options).then(function (body) {
+                        var tokenInfo = JSON.parse(body); // This includes refresh token, scope etc..
+                        var getOptions = {
+                            url: 'https://api.meethue.com/v2/bridges',
+                            method: "GET",
+                            headers: {
+                                'Authorization': 'Bearer ' + tokenInfo.access_token
+                            },
+                        };
+
+                        console.log('Getting bridge ID associated to the account...');
+
+                        return request(getOptions)
+                            .then(function (body) {
+
+                                //Get bridgeId
+                                var bridgeIds = JSON.parse(body);
+                                if (bridgeIds.length > 0) {
+
+                                    console.log('Success.');
+
+                                    var putOptions = {
+                                        url: 'https://api.meethue.com/v2/bridges/' + bridgeIds[0].id + '/0/config',
+                                        method: "PUT",
+                                        headers: {
+                                            'Authorization': 'Bearer ' + tokenInfo.access_token,
+                                            'Content-Type': 'application/json'
+                                        },
+                                        json: { 'linkbutton': true }
+                                    };
+
+                                    console.log('Getting whitelist ID from bridge ' + bridgeIds[0].id + '...');
+
+                                    return request(putOptions)
+                                        .then(function (body) {
+                                            //pressed link button
+                                            if (typeof body[0].success !== 'undefined') {
+                                                var postOptions = {
+                                                    url: 'https://api.meethue.com/v2/bridges/' + bridgeIds[0].id + '/',
```

Comment:
>Ids[0] [](start = 103, length = 6)

More a question for opent2t folks - Should we do some standard parameter validation here? eg checking whether the array contains the expected number of elements so we dont accidentally run into IndexOutOfBound type exceptions? In general though as long as we log the error and report a 500(internal server error) as our default, shouldnt matter but just wanted to also what sorts of error handling translators are expected to be doing